src/functions-and-arrays.js

```diff
@@ -1,41 +1,147 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(number1,number2) {
+  if (number1 > number2){
+    return number1;
+  }
+  else if (number1 < number2) {
+    return number2;
+  }
+  else {
+    return number1;
+  }
+}
+maxOfTwoNumbers(3,56)
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
+function findLongestWord(longestWord) {
+    if (longestWord.length === 0){
+      return null;
+    }
+    else if (longestWord.length === 1){
+      return longestWord[0];
+    }
+    let result = "";
+    for (let i= 0 ; i <longestWord.length; i++) {
+      if (longestWord[i].length > result.length) {
+        result = longestWord[i];
+      }
+    }
+    return result;
+  };
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
-
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  if (numbers.length === 0){
+      return 0;
+  }
+  let sum = 0;
+  for (let i=0; i<numbers.length ; i++){
+    if (numbers.length === 1){
+      return numbers[i];
+    }
+    else if (numbers[i] === 0){
+      return 0;
+    }
+    else {
+      sum += numbers[i];
+    }
+  }
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(elements) {
+  if (elements.length === 0){
+    return 0;
+  }
+  let total = 0;
+  for (let i=0; i<elements.length ; i++){
+    if (elements[i] === 0){
+      return 0;
+    }
+    else if (elements.length === 1){
+      return elements[i];
+    }
+    else {
+      total += elements[i] || elements[i].length;
+   
+    }
+  }
+  return total;
+
+
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+
+
+
+
+function averageNumbers(theNumbers) {
+  let total = 0;
+  if (theNumbers.length === 0){
+    return null;
+  }
+  for (let i= 0; i < theNumbers.length ; i++){
+    if (theNumbers.length === 1){
+      return theNumbers[i];
```

Comment:
Since you wrote earlier a function to sum numbers, you can reuse it here, instead of repeating the whole logic to sum the numbers again. By calling the function you can do that.
Something like this: 
```
let sum = sumNumbers(theNumbers)
return sum / theNumbers.length
```